processes/developmentSetup.py

```diff
@@ -11,136 +11,169 @@
-
-        self.fetchHathiSampleData()
-
-        procArgs = ['complete'] + ([None] * 4)
-
-        self.createRedisClient()
-        self.clear_cache()
-
-        classifyProc = ClassifyProcess(*procArgs)
-        classifyProc.runProcess()
-
-        catalogProc = CatalogProcess(*procArgs)
-        catalogProc.runProcess(max_attempts=1)
+        try:
+            self.generateEngine()
+            self.createSession()
+
+            self.initializeDatabase()
+
+            self.createElasticConnection()
+            self.wait_for_elastic_search()
+            self.createElasticSearchIndex()
+
+            self.createRabbitConnection()
+            self.createOrConnectQueue(os.environ['OCLC_QUEUE'], os.environ['OCLC_ROUTING_KEY'])
+            self.createOrConnectQueue(os.environ['FILE_QUEUE'], os.environ['FILE_ROUTING_KEY'])
+
+            self.fetch_hathi_sample_data()
+
+            process_args = ['complete'] + ([None] * 4)
+
+            self.createRedisClient()
+            self.clear_cache()
+
+            classify_process = ClassifyProcess(*process_args)
+            classify_process.runProcess()
+
+            catalog_process = CatalogProcess(*process_args)
+            catalog_process.runProcess(max_attempts=1)
+            
+            cluster_process = ClusterProcess(*process_args)
+            cluster_process.runProcess()
+        except Exception:
+            logger.exception(f'Failed to run development setup process')
+
+    def initialize_db(self):
+        self.admin_db_manager.generateEngine()
+
+        with self.admin_db_manager.engine.connect() as admin_db_connection:
+            admin_db_connection.connection.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
+
+            self.create_database(admin_db_connection)
+            self.create_database_user(admin_db_connection)
+
+        self.admin_db_manager.engine.dispose()
+
+    def create_database(self, db_connection):
+        try:
+            db_connection.execute(
+                sa.text(f"CREATE DATABASE {os.environ['POSTGRES_NAME']}"),
+            )
+        except ProgrammingError:
+            pass
+        except Exception as e:
+            logger.exception('Failed to create database')
+            raise e
+
+    def create_database_user(self, db_connection):
+        try:
+            db_connection.execute(
+                sa.text(
+                    f"CREATE USER {os.environ['POSTGRES_USER']} "
+                    f"WITH PASSWORD '{os.environ['POSTGRES_PSWD']}'",
+                ),
+            )
+            db_connection.execute(
+                sa.text(
+                    f"GRANT ALL PRIVILEGES ON DATABASE {os.environ['POSTGRES_NAME']} "
+                    f"TO {os.environ['POSTGRES_USER']}",
+                ),
+            )
+        except ProgrammingError:
+            pass
+        except Exception as e:
+            logger.exception('Failed to create database user')
+            raise e
```

Comment:
Good idea to separate those lines of code into two separate methods for clarity.